app/questionnaire/questionnaire_store_updater.py

```diff
@@ -81,6 +100,31 @@ def remove_answer_ids(self, answer_ids: List):
             self._answer_store.remove_answer(answer_id)
         self._questionnaire_store.add_or_update()
 
-    def remove_completed_blocks(self, location):
-        self._questionnaire_store.remove_completed_blocks(location)
+    def remove_completed_block(self, location):
+        self._questionnaire_store.completed_items.block_locations.remove(location)
+        self._questionnaire_store.add_or_update()
+
+    def add_completed_section(self, section_id):
+        self._questionnaire_store.completed_items.sections.append(section_id)
+
+    def remove_completed_section(self, section_id):
+        self._questionnaire_store.completed_items.sections.remove(section_id)
+
+        section_summary_id = self._schema.get_section_summary(section_id)
+        if section_summary_id:
+            self._questionnaire_store.completed_items.block_locations.remove(
```

Comment:
This makes a completed section incomplete if you were to visit any block within the completed section. Which isn't ideal. Due to this behavior, this is causing some 500s when clicking back/using change links, etc. Happy to discuss the behaviors.

This would also cause weird behavior with hub and spoke as the user may visit a block to view details but then return to the hub. A section is only marked a complete on a POST request from the section summary (last block).